docs/src/main/asciidoc/cdi-integration.adoc

```diff
@@ -0,0 +1,597 @@
+
+_Solution_: Use the `AdditionalBeanBuildItem`. 
+This build item can be used to specify one or more additional classes to be analyzed during the discovery. 
+Additional bean classes are transparently added to the application index processed by the container.
+
+.`AdditionalBeanBuildItem` Example
+[source,java]
+----
+@BuildStep
+AdditionalBeanBuildItem additionalBeans() {
+     return new AdditionalBeanBuildItem(SmallRyeHealthReporter.class, HealthServlet.class)); <1>
+}
+----
+<1> Convenient `AdditionalBeanBuildItem.Builder` can be used for more complex use cases.
+
+Bean classes added via `AdditionalBeanBuildItem` are _removable_ by default.
+If the container considers them <<cdi-reference.adoc#remove_unused_beans,unused>> they are just ignored.
+However, you can use `AdditionalBeanBuildItem.Builder.setUnremovable()` method to instruct the container to never remove bean classes registered via this build item.
+See also <<cdi-reference.adoc#remove_unused_beans,Removing Unused Beans>> and <<unremovable_builditem>> for more details.
+
+It is aso possible to set the default scope via `AdditionalBeanBuildItem.Builder#setDefaultScope()`.
+The default scope is only used if there is no scope declared on the bean class. 
+
+NOTE: If no default scope is specified the `@Dependent` pseudo-scope is used.
+
+=== _Reason 2_: Class Is Discovered but Has No Bean Defining Annotation
+
+In Quarkus, the application is represented by a single bean archive with the https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#default_bean_discovery[bean discovery mode `annotated`, window="_blank"].
+Therefore, bean classes that don't have a http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#bean_defining_annotations[bean defining annotation, window="_blank"] are ignored.
+Bean defining annotations are declared on the class-level and incluce scopes, stereotypes and `@Interceptor`.
+
+_Solution 1_: Use the `AutoAddScopeBuildItem`. This build item can be used to add a scope to a class that meets certain conditions.
+
+.`AutoAddScopeBuildItem` Example
+[source,java]
+----
+@BuildStep
+AutoAddScopeBuildItem autoAddScope() {
+   return AutoAddScopeBuildItem.builder().containsAnnotations(SCHEDULED_NAME, SCHEDULES_NAME) <1>
+      .defaultScope(BuiltinScope.SINGLETON) <2>
+      .build();
+}
+----
+<1> Find all classes annotated with `@Scheduled`.
+<2> Add `@Singleton`. Classes annotated with a scope are skipped automatically.
+
+_Solution 2_: If you need to process classes annotated with a specific annotation then it's possible to extend the set of bean defining annotations via the `BeanDefiningAnnotationBuildItem`.
+
+.`BeanDefiningAnnotationBuildItem` Example
+[source,java]
+----
+@BuildStep
+BeanDefiningAnnotationBuildItem additionalBeanDefiningAnnotation() {
+   return new BeanDefiningAnnotationBuildItem(Annotations.GRAPHQL_API); <1>
+}
+----
+<1> Add `org.eclipse.microprofile.graphql.GraphQLApi` to the set of bean defining annotations.
```

Comment:
In such case, what's the scope of the bean?

Reply:
It should be the same as with `AdditionalBeanBuildItem`, i.e. `@Dependent`. I'll add a note...